### PR TITLE
KEP-843, KEP-845: Update crud script to handle new pbft response style,

### PR DESCRIPTION
**Pull request by isabelsavannah**
fix some lingering bugs that prevented that response from being sent

the session is held as a shared_ptr instead of a weak_ptr because, between pbft's ack and crud's response, no operations are pending on that session and it will otherwise be cleaned up.

### CODE REVIEW

File: pbft/pbft_operation.cpp [hunk 1]
```diff
@@ -193,13 +193,13 @@ pbft_operation::debug_string() const
 }
 
 void
-pbft_operation::set_session(std::weak_ptr<bzn::session_base> session)
+pbft_operation::set_session(std::shared_ptr<bzn::session_base> session)
```
Comment (ebruck) [inline pbft/pbft_operation.cpp]: We want weak pointers to not hold a reference to a connection that has gone away.

Comment (ebruck) [reply pbft/pbft_operation.cpp]: There should always be a read scheduled.

Comment (isabelsavannah) [reply pbft/pbft_operation.cpp]: There is no read scheduled between when pbft receives the database message and when the response is sent; send_message is what re-schedules the read. 

If we _do_ keep the session alive by immediately scheduling another read (this was previously happening as a side effect of pbft's early ack response), then when we try to send the actual response later, we crash on an assert inside boost with a comment saying "you can't have multiple pending async operations". We would have to intentionally schedule a (fake) read and the explicitly cancel it later (which session_base doesn't presently even allow), at which point it the read just feels like a nonobvious way to force the session to stay alive. The shared_ptr is owned by the pbft_operation anyway, which will automatically be cleaned based on the checkpoint process.

Comment (ebruck) [reply pbft/pbft_operation.cpp]: Can you add a way to ask the session if the connection is still open? That way I can still bypass expensive db reads if no one is around to receive them.

Comment (ebruck) [reply pbft/pbft_operation.cpp]: If you are getting a multiple read's scheduled then you don't have to do anything.

Comment (isabelsavannah) [reply pbft/pbft_operation.cpp]: Maybe multiple reads work, but if you try to write while a read is outstanding it crashes on that assert.

"multiple reads scheduled" - reads on the database or on the socket?

**Pull request merged**